lib/tasks/term_generation/calculate_project_statistics.rake

```diff
@@ -0,0 +1,35 @@
+namespace :term_generation do
+  desc "Calculate statistics for a Term Generation project"
+  task :calculate_project_statistics, [:project_id] => :environment do |_, args|
+    project = TaxonomyProject.find(args.project_id)
```

Comment:
You could extract the logic of updating the statistics to a separate service object or maybe the model itself so you can write unit tests